### PR TITLE
Removes a Dreamchecker check and remove pointless return

### PR DESCRIPTION
**Pull request by redmoogle**
### Intent of your Pull Request

Prunes useless errors from the codebase

### Why is this good for the game?

Makes it easier to see actual problems

### CODE REVIEW

**Review by jcatjmeow — APPROVED** [review]

**Comment by Bibby0110** [comment]
Why are you removing SHOULD_BE_PURE from CanPass?

File: code/game/atoms.dm [hunk 1]
```diff
@@ -225,7 +225,6 @@
 ///Can the mover object pass this atom, while heading for the target turf
 /atom/proc/CanPass(atom/movable/mover, turf/target)
 	SHOULD_CALL_PARENT(TRUE)
```
Comment (Bibby0110) [inline code/game/atoms.dm:227]:
```suggestion
	SHOULD_CALL_PARENT(TRUE)
	SHOULD_BE_PURE(TRUE)
```

File: code/game/atoms.dm [hunk 2]
```diff
@@ -236,7 +235,6 @@
 /// Returns true or false to allow the mover to move through src
 /atom/proc/CanAllowThrough(atom/movable/mover, turf/target)
 	SHOULD_CALL_PARENT(TRUE)
```
Comment (Bibby0110) [inline code/game/atoms.dm:237]:
```suggestion
	SHOULD_CALL_PARENT(TRUE)
	//SHOULD_BE_PURE(TRUE) //Until we fix the numerous cases where this isn't being followed
```

**Review by Bibby0110 — COMMENTED** [review]
💋

**Comment by redmoogle** [comment]
> 
> 
> 💋

Nah, nothing of value lossed

**Comment by alexkar598** [comment]
Why? What's wrong with SHOULD_BE_PURE?

**Comment by redmoogle** [comment]
> Why? What's wrong with SHOULD_BE_PURE?

Generates annoying 'shitcode errors' which who tf is gonna fix

**Comment by redmoogle** [comment]
Pointless leaving it open as alexkar has his PR open ~~not that I was working on that first >:(~~

**Pull request closed without merging**